framework/decode/openxr_replay_consumer_base.cpp

```diff
@@ -685,10 +686,19 @@ void OpenXrReplayConsumerBase::Process_xrPollEvent(const ApiCallInfo&
                         received_events_.erase(received_events_.begin(), received_events_.begin() + 100);
                     }
                 }
-                else
+                else if (replay_result == XR_EVENT_UNAVAILABLE)
                 {
                     // Yield and retry
-                    std::this_thread::sleep_for(std::chrono::nanoseconds(1));
+                    std::this_thread::sleep_for(std::chrono::nanoseconds(sleep_time));
+
+                    // Next time, sleep for double what we initially slept for.  This way if we're just
+                    // spinning, we spin less and less each time.
+                    sleep_time *= 2;
```

Comment:
Should add a max sleep of 500ms or so, and reduce the retry limit s.t. the maximum wait is < something like 5-10 seconds total.